### PR TITLE
Refactors payoff.py and cooperation.py

### PR DESCRIPTION
**Pull request by meatballs**
1. Refactors payoff_matrix to use interactions dictionary instead of matrix
2. Adds cooperation_matrix function to return cooperation matrix from interactions dictionary
3. Tidies up docstrings in both files

### CODE REVIEW

**Comment by drvinceknight** [comment]
This looks good to me, could you make a note on #372 re the slight change that will be needed (nice to include the cooperation_matrix) to http://axelrod.readthedocs.org/en/299/tutorials/further_topics/morality_metrics.html

(Just trying to keep track).

:+1:

File: axelrod/cooperation.py [hunk 1]
```diff
@@ -1,7 +1,66 @@
...
+
+        i.e. an n by n matrix where n is the number of players. Each row (i)
+        and column (j) represents an individual player and the the value Cij
+        is the number of times player i cooperated against opponent j.
     """
+    cooperation = [[0 for i in range(nplayers)] for j in range(nplayers)]
+    for players, actions in interactions.items():
+        p1_actions, p2_actions = zip(*actions)
+        p1_cooperation = p1_actions.count(Actions.C)
+        p2_cooperation = p2_actions.count(Actions.C)
```
Comment (CamDavidsonPilon) [inline axelrod/cooperation.py]: You can just use `C` and `D` here

File: axelrod/cooperation.py [hunk 2]
```diff
@@ -1,7 +1,66 @@
 from . import eigen
+from axelrod import Actions
 
-def cooperation_matrix(results):
+C, D = Actions.C, Actions.D
+
+# As yet unused until RoundRobin returns interactions
+def cooperation_matrix(interactions, nplayers):
```
Comment (CamDavidsonPilon) [inline axelrod/cooperation.py]: given an `interactions` dict, couldn't the number of players be inferred? Basically solving for `len(dict.keys()) = n*(n-1)/2`

Comment (meatballs) [inline axelrod/cooperation.py]: Good call. That's me being lazy! I'll add it in.

File: axelrod/cooperation.py (hunk 1 above)
Comment (meatballs) [inline axelrod/cooperation.py]: Ah yes! I added the variables in at the top of the file and then forgot to use them. Duh.

**Comment by meatballs** [comment]
I'm done messing with this for the moment. Feel free to comment/merge/criticise!

**Comment by marcharper** [comment]
:+1: to merge

**Pull request merged**